### PR TITLE
Use hqDefine and hqImport for keyboard shortcuts

### PR DESCRIPTION
**Pull request by orangejenny**
Get rid of some globals.

I'm using the convention `hqImport('my/html/file.html#moduleName')` for modules that aren't convenient to move into their own js files.

@dannyroberts / @gcapalbo

### CODE REVIEW

File: corehq/apps/hqwebapp/static/hqwebapp/js/keyboard_navigator.js [hunk 1]
```diff
@@ -1,182 +1,177 @@
...
-        };
-
-        self.gen_handle_nav = function(key) {
-            return function() {
-                if (!self.navigating) {
-                    self.enter_nav();
+            };
+    
+            self.leave_context = function(side) {
+                self.set_index(side === 'end' ? 0 : self.$elements.length - 1)
```
Comment (dimagimon) [inline corehq/apps/hqwebapp/static/hqwebapp/js/keyboard_navigator.js]: Missing semicolon.

File: corehq/apps/hqwebapp/static/hqwebapp/js/keyboard_navigator.js [hunk 2]
```diff
@@ -1,182 +1,177 @@
...
+                return function() {
+                    if (!self.navigating) {
+                        self.enter_nav();
+                    }
+                    self.navigating = true;
+                    self.handle_focus_out();
+                    self.set_index(self.index + (self.forward_keys.indexOf(character) > -1 ? 1 : -1));
+                    self.handle_focus_in();
+                    return false;
+                }
```
Comment (dimagimon) [inline corehq/apps/hqwebapp/static/hqwebapp/js/keyboard_navigator.js]: Missing semicolon.

File: corehq/apps/hqwebapp/static/hqwebapp/js/keyboard_navigator.js [hunk 3]
```diff
@@ -1,182 +1,177 @@
...
+            };
+    
+            self.handle_action = function() {
+                // last_mover is a pseudo-global variable stored in a closure that keeps track of which kn moved last
+                // only want to handle the action of this navigator if it was the last mover.
+                if (last_mover === self.name) {
+                    self.action_fn(self.$active_element);
+                    self.leave_nav();
+                }
+                return false
```
Comment (dimagimon) [inline corehq/apps/hqwebapp/static/hqwebapp/js/keyboard_navigator.js]: Missing semicolon.

**Comment by gcapalbo** [comment]
looks good to me, does this need qa or is it good to go?

also looks like there are conflicts

**Comment by dannyroberts** [comment]
I like the initiative in moving things over to hqDefine! I'm seeing this problem on staging:

screen shot 2016-02-29 at 5 38 55 pm

**Comment by orangejenny** [comment]
Merge conflict resolved. I think this is fine without QA - it should fail obviously, and I've tested what I can think of (B2 pages, B3 pages, app manager, users with shortcuts enabled/disabled).

@dannyroberts turns out that's a bug on prod - thanks for giving a me a head start on fixing it before users started noticing

**Comment by gcapalbo** [comment]
@orangejenny ok, sorry i just want to make sure - do we need to fix the issue Danny pointed out before we merge this?

**Comment by orangejenny** [comment]
@gcapalbo It's been fixed in prod (https://github.com/dimagi/commcare-hq/pull/10555)

**Comment by gcapalbo** [comment]
ok thanks

**Pull request merged**